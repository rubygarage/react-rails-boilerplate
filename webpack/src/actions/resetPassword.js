import { SEND_RESTORE_PASSWORD_EMAIL, REQUEST } from 'constants/actions'

export function sendRestorePasswordEmail(values, resolve, reject) {
  return { type: SEND_RESTORE_PASSWORD_EMAIL + REQUEST, values, resolve, reject }
}
