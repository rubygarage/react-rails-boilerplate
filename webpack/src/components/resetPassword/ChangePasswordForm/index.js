import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedMessage, intlShape } from 'react-intl';
import { Field } from 'redux-form';
import { Link } from 'react-router-dom';
import ErrorMessage from 'components/ui/ErrorMessage';

import styles from './styles.css';

export function ChangePasswordForm({
  handleSubmit,
  pristine,
  submitHandler,
  submitting,
  intl: { formatMessage },
}) {
  return (
    <div className="container">
      <div className="row">
        <div className="col-md-6 offset-md-3">
          <div className={styles['register-wrap']}>
            <div className="card">
              <div className="card-header text-center">
                <h2> <FormattedMessage id="restore_password.header" /> </h2>
              </div>
              <div className="card-block">
                <form onSubmit={handleSubmit(submitHandler)}>
                  <ErrorMessage name="password" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password"
                      placeholder={formatMessage({ id: 'restore_password.new_password' })}
                      aria-label={formatMessage({ id: 'restore_password.new_password' })}
                    />
                  </div>

                  <ErrorMessage name="password_confirmation" />
                  <div className="input-group mb-3">
                    <Field
                      type="password"
                      component="input"
                      className="form-control"
                      name="password_confirmation"
                      placeholder={formatMessage({ id: 'restore_password.confirm_password' })}
                      aria-label={formatMessage({ id: 'restore_password.confirm_password' })}
                    />
                  </div>

                  <button className="btn btn-primary mb-3" type="submit" disabled={pristine || submitting}>
                    <FormattedMessage id="restore_password.change_my_password" />
                  </button>

                </form>

                <Link className="mr-3" to="/sign_up">
                  <FormattedMessage id="sign_in.register" />
                </Link>

                <Link className="mr-3" to="/sign_in">
                  <FormattedMessage id="sign_up.login" />
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

ChangePasswordForm.propTypes = {
  intl: intlShape,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitHandler: PropTypes.func.isRequired,
  submitting: PropTypes.bool,
};

export default injectIntl(ChangePasswordForm);
