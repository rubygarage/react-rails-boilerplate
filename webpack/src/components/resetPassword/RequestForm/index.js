import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage, intlShape } from 'react-intl'
import { Field } from 'redux-form'
import ErrorMessage from 'components/ui/ErrorMessage'

import styles from './styles.css'

export function ResetPasswordRequestForm({
  handleSubmit,
  pristine,
  submitHandler,
  submitting,
  intl: { formatMessage }
}) {
  return (
    <div className="container">
      <div className="row">

        <div className="col-md-6 offset-md-3">
          <div className={styles['register-wrap']}>
            <div className="card">
              <div className="card-header text-center">
                <h2> <FormattedMessage id="restore_password.header" /> </h2>
              </div>
              <div className="card-block">
                <form onSubmit={handleSubmit(submitHandler)}>

                  <ErrorMessage name="email" />
                  <div className="input-group mb-3">
                    <Field
                      type="text"
                      component="input"
                      className="form-control"
                      name="email"
                      placeholder={formatMessage({ id: 'restore_password.your_email' })}
                      aria-label={formatMessage({ id: 'restore_password.your_email' })}
                    />
                  </div>

                  <button className="btn btn-primary mb-3" type="submit" disabled={pristine || submitting}>
                    <FormattedMessage id="restore_password.send_instructions" />
                  </button>

                </form>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  )
}

ResetPasswordRequestForm.propTypes = {
  intl: intlShape,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool,
  submitHandler: PropTypes.func.isRequired,
  submitting: PropTypes.bool
}

export default injectIntl(ResetPasswordRequestForm)
