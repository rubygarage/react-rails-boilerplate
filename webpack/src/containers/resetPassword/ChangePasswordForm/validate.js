const validate = (values, state) => {
  const { formatMessage } = state.intl;
  const errors = {};

  if (!values.password) {
    errors.password = formatMessage({ id: 'validation.required_field' }, { field: 'Password' });
  }

  if (!values.password_confirmation) {
    errors.password_confirmation = formatMessage({ id: 'validation.required_field' }, { field: 'Password confirmation' });
  } else if (values.password_confirmation !== values.password) {
    errors.password_confirmation = formatMessage({ id: 'validation.password_confirmation_does_not_match_the_password' });
  }

  return errors;
};

export default validate;
