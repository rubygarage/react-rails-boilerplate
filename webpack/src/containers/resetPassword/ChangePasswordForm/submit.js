import { parseJsonApiErrors } from 'utils/validation'
import { SubmissionError } from 'redux-form'
import { UPDATE_PASSWORD, REQUEST } from 'constants/actions'

const submit = (values, dispatch, props) => new Promise((resolve, reject) => {
  dispatch({ type: UPDATE_PASSWORD + REQUEST, values, resolve, reject })
}).catch((response) => {
  const errors = parseJsonApiErrors(response.errors, props.fieldsForValidation)
  throw new SubmissionError(errors)
})

export default submit
