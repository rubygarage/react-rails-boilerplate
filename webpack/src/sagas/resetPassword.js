import ApiClient from 'utils/apiClient';
import { takeEvery, call, put } from 'redux-saga/effects';
import { SEND_RESTORE_PASSWORD_EMAIL, UPDATE_PASSWORD, REQUEST, SUCCESS, ERROR } from 'constants/actions';
import redirect from 'helpers/redirect';

export function* sendRestorePasswordEmail({ values, resolve, reject }) {
  const apiClient = new ApiClient().buildClient();

  try {
    yield call(apiClient.post, '/api/v1/auth/users/password', values);
    yield put({ type: SEND_RESTORE_PASSWORD_EMAIL + SUCCESS });
    yield call(resolve);
  } catch (error) {
    yield put({ type: SEND_RESTORE_PASSWORD_EMAIL + SUCCESS, error });
    yield call(reject, error.response.data);
  }
}

export function* validateToken(token, res) {
  const apiClient = new ApiClient().buildClient();

  try {
    yield call(apiClient.get, `/api/v1/auth/users/password?reset_token=${token}`);
  } catch (error) {
    yield call(redirect, '/404', res);
  }
}

export function* updatePassword({ values, resolve, reject }) {
  const apiClient = new ApiClient().buildClient();

  try {
    yield call(apiClient.put, '/api/v1/auth/users/password', values);
    yield put({ type: UPDATE_PASSWORD + SUCCESS });
    yield call(resolve);
    yield call(redirect, '/sign_in');
  } catch (error) {
    yield put({ type: UPDATE_PASSWORD + ERROR, error });
    yield call(reject, error.response.data);
  }
}

export function* watchSendRestorePassword() {
  yield takeEvery(SEND_RESTORE_PASSWORD_EMAIL + REQUEST, sendRestorePasswordEmail);
}

export function* watchUpdatePassword() {
  yield takeEvery(UPDATE_PASSWORD + REQUEST, updatePassword);
}
