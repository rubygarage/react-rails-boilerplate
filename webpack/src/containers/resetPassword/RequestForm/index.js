import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import { injectIntl } from 'react-intl';
import sendRestorePasswordEmail from 'actions/resetPassword';

import Instructions from 'components/resetPassword/RequestForm/Instructions';
import ResetPasswordRequestFormComponent from 'components/resetPassword/RequestForm';

import validate from './validate';
import submit from './submit';

export class ResetPasswordRequestForm extends Component {
  static propTypes = {
    sendRestorePasswordEmail: PropTypes.func.isRequired,
  }

  state = {
    resetRequestSubmitted: false,
  }

  setResetRequestSubmitted = (value) => {
    this.setState({ resetRequestSubmitted: value });
  };

  submitAndHandleStateChange = (values, dispatch, props) => {
    submit(values, dispatch, { ...props, setResetRequestSubmitted: this.setResetRequestSubmitted });
  }

  render() {
    const props = { ...this.props, submitHandler: this.submitAndHandleStateChange };
    const component = this.state.resetRequestSubmitted
      ? <Instructions />
      : <ResetPasswordRequestFormComponent {...props} />;
    return (component);
  }
}

const mapDispatchToProps = { sendRestorePasswordEmail };

export default connect(null, mapDispatchToProps)(injectIntl(reduxForm({
  form: 'resetPassword',
  fieldsForValidation: ['email'],
  validate,
})(ResetPasswordRequestForm)));
