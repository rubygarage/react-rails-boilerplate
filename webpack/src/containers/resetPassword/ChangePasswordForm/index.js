import React, { Component } from 'react'
import { reduxForm } from 'redux-form'
import { connect } from 'react-redux'
import submit from './submit'
import { validateToken } from 'sagas/resetPassword'

import ChangePasswordFormComponent from 'components/resetPassword/ChangePasswordForm'

export class ChangePasswordForm extends Component {
  render() {
    const props = {
      ...this.props,
      submitHandler: submit
    }

    return (
      <ChangePasswordFormComponent {...props} />
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  const { reset_password_token } = ownProps.match.params

  return ({
    initialValues: { reset_token: reset_password_token }
  })
}

function preload(params, req, res) {
  const sagasToComplete = []
  const { reset_password_token } = params

  sagasToComplete.push([validateToken, reset_password_token, res]) // eslint-disable-line

  return sagasToComplete
}
ChangePasswordForm.preload = preload

export default connect(mapStateToProps)(reduxForm({
  form: 'changePassword',
  fieldsForValidation: ['password']
})(ChangePasswordForm))
