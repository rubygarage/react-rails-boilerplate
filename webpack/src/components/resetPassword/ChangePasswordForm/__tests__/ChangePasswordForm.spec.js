import React from 'react';
import { shallow } from 'enzyme';
import { ChangePasswordForm } from 'components/resetPassword/ChangePasswordForm';

describe('<ChangePasswordForm />', () => {
  let props = {
    intl: { formatMessage: jest.fn() },
    handleSubmit: jest.fn(),
    submitHandler: jest.fn(),
  };

  it('renders ChangePasswordForm component', () => {
    const component = shallow(<ChangePasswordForm {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders with pristine = true', () => {
    props = {
      ...props,
      pristine: true,
    };

    const component = shallow(<ChangePasswordForm {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders with submitting = true', () => {
    props = {
      ...props,
      submitting: true,
    };

    const component = shallow(<ChangePasswordForm {...props} />);

    expect(component).toMatchSnapshot();
  });
});
