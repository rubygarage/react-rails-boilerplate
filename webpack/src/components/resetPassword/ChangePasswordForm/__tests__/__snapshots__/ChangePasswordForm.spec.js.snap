// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePasswordForm /> renders ChangePasswordForm component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="restore_password.header"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                type="submit"
              >
                <FormattedMessage
                  id="restore_password.change_my_password"
                  values={Object {}}
                />
              </button>
            </form>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_up"
            >
              <FormattedMessage
                id="sign_in.register"
                values={Object {}}
              />
            </Link>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_in"
            >
              <FormattedMessage
                id="sign_up.login"
                values={Object {}}
              />
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChangePasswordForm /> renders with pristine = true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="restore_password.header"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="restore_password.change_my_password"
                  values={Object {}}
                />
              </button>
            </form>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_up"
            >
              <FormattedMessage
                id="sign_in.register"
                values={Object {}}
              />
            </Link>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_in"
            >
              <FormattedMessage
                id="sign_up.login"
                values={Object {}}
              />
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ChangePasswordForm /> renders with submitting = true 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <div
        className="register-wrap"
      >
        <div
          className="card"
        >
          <div
            className="card-header text-center"
          >
            <h2>
               
              <FormattedMessage
                id="restore_password.header"
                values={Object {}}
              />
               
            </h2>
          </div>
          <div
            className="card-block"
          >
            <form>
              <ErrorMessage
                name="password"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password"
                  type="password"
                />
              </div>
              <ErrorMessage
                name="password_confirmation"
              />
              <div
                className="input-group mb-3"
              >
                <Field
                  className="form-control"
                  component="input"
                  name="password_confirmation"
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary mb-3"
                disabled={true}
                type="submit"
              >
                <FormattedMessage
                  id="restore_password.change_my_password"
                  values={Object {}}
                />
              </button>
            </form>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_up"
            >
              <FormattedMessage
                id="sign_in.register"
                values={Object {}}
              />
            </Link>
            <Link
              className="mr-3"
              replace={false}
              to="/sign_in"
            >
              <FormattedMessage
                id="sign_up.login"
                values={Object {}}
              />
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
